flake.nix

```diff
@@ -0,0 +1,147 @@
+{
+  description = "The More Aggressive Nix Eval Cache";
+
+  inputs = {
+    flakeCompat.url = github:edolstra/flake-compat;
+    flakeCompat.flake = false;
+
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = inputs: let
+    commit = inputs.self.shortRev or "dirty";
+    date = inputs.self.lastModifiedDate or inputs.self.lastModified or "19700101";
+    version = "1.2.0+${builtins.substring 0 8 date}.${commit}";
+
+    nixpkgsForHost = host:
+      import inputs.nixpkgs {
+        overlays = [
```

Comment:
Is this overlay used anywhere? I would avoid an overlay to not having to re-import nixpkgs for better eval performance: https://zimbatm.com/notes/1000-instances-of-nixpkgs

Reply:
Iirc, the reason is to get the cross compile machinery. I was having similar concerns but the cross machinery is not well exposed in upstream nixpkgs.

Definitely something to look for, though.